stix_shifter_modules/aws_athena/stix_transmission/query_connector.py

```diff
@@ -57,36 +62,72 @@ async def create_query_connection(self, query):
-                query = query_string.replace('UNION (', 'UNION ({}'.format(select_statement)).\
+                query_with_select = query_string.replace('UNION (', 'UNION ({}'.format(select_statement)).\
                     replace('INTERSECT (', 'INTERSECT ({}'.format(select_statement))
             else:
-                query = select_statement + query[query_service_type]
+                query_with_select = select_statement + query[query_service_type]
             result_config = self.get_result_config()
-            query_args = {"QueryString": query, "ResultConfiguration": result_config}
-            response_dict = await self.client.makeRequest('athena', 'start_query_execution', **query_args)
-            return_obj['success'] = True
-            return_obj['search_id'] = response_dict['QueryExecutionId'] + ":" + query_service_type
-
+            
+            return await self.query_api(query_with_select, result_config, query_service_type, return_obj)
+            
         except Exception as ex:
             response_dict['__type'] = ex.__class__.__name__
             response_dict['message'] = ex
             ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
         return return_obj
 
-    async def column_list(self, table):
+    async def query_api(self, query, result_config, query_service_type, return_obj):
+        """Creates a query job and ensures that none of the columns requested are missing from the query.
+
+        Args:
+            query (String): The original query without modification. 
+            result_config (Dict): Query configuration.
+            query_service_type (String): The type of query ("OCSF", "VPCFlow, etc)
+            return_obj (Dict): Contains the metadata about the request.
+
+        Returns:
+            Dict: The metadata about the request such as the query/search ID and the status.
+        """
+        logger.debug(f"The current query is : {query}")
+
+        #Creates the initial query job.
+        query_args = {"QueryString": query, "ResultConfiguration": result_config}
+        response_dict = await self.client.makeRequest('athena', 'start_query_execution', **query_args)
+        return_obj['search_id'] = response_dict['QueryExecutionId'] + ":" + query_service_type
+
+        modified_query = dict()
+        modified_query = await PostQueryConnectorErrorHandler.check_status_for_missing_column(self.client, return_obj['search_id'], query)
+        #If the query is successful (or the exception isn't column related) than it's considered a success and exits.
+        #If 10 columns are not found or it fails to replace a column 10 times, than it exits (to prevent endless loops).
+        #If the query is not successful, than it will retry the query with the modified query.
+        if(modified_query == "CONNECTOR_FACTORY_SUCCESS" or self.total_try_count > 10):
+            logger.debug(f"The total attempt count was {self.total_try_count}")
```

Comment:
Add a text that indicates what it is trying to attempt so that it is easily understandable from the log